### PR TITLE
feat: remove injection options

### PR DESCRIPTION
**Pull request by Rauno56**
# Description

- POR-9619

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Added automated tests
- [x] Unit tests have been added/updated
- [ ] Documentation has been updated
- [x] Change file has been generated (`npm run change:new`)
- [ ] Delete this branch (after the PR is merged)

### CODE REVIEW

File: examples/log-injection/index.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// This will run a function inside a context which has an active span.
+context.with(spanContext, doWork);
+
+// Even though the span has not ended yet it's not active in current context
+// anymore and thus will not be logged.
+log.info('done!');
+span.end();
+
+setTimeout(() => {
+	// wait for the spans to be flushed
```
Comment (jtmal-signalfx) [inline examples/log-injection/index.js:28]: what about ForceFlush? [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#forceflush) [implementation](https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-sdk-trace-base/src/BasicTracerProvider.ts#L152)

File: test/loginjection.test.ts [hunk 2]
```diff
@@ -15,32 +15,41 @@
...
+import { PinoInstrumentation } from '@opentelemetry/instrumentation-pino';
 
 describe('log injection', () => {
   let stream: Writable;
   let record: any;
 
-  function assertInjection(logger, done, extra?) {
+  function assertInjection(logger, extra?) {
     const span = trace.getTracer('test').startSpan('main');
+    extra = extra ?? [['service.name', 'test-service']];
```
Comment (jtmal-signalfx) [inline test/loginjection.test.ts:34]: `function assertInjection(logger, extra = [['service.name', 'test-service']]) {`

File: test/loginjection.test.ts [hunk 3]
```diff
@@ -54,32 +63,83 @@ describe('log injection', () => {
...
+    });
+  });
+
+  describe('injecting with custom hook', () => {
+    afterEach(() => {
+      stopTracing();
+    });
+
+    it('is possible to opt out from injecting resource attributes', () => {
+      const MY_VALUE = 'myValue';
```
Comment (jtmal-signalfx) [inline test/loginjection.test.ts:101]: We have more than one test case using the same values, let's randomise or make these different for each test case.

File: test/loginjection.test.ts [hunk 4]
```diff
@@ -54,32 +63,83 @@ describe('log injection', () => {
 
   describe('default flow', () => {
```
Comment (jtmal-signalfx) [inline test/loginjection.test.ts:64]: Is there a test for disabled log injection?

**Comment by jtmal-signalfx** [comment]
I unchecked the "documentation is done", because there's this sentence in the README:
> If log injection is enabled, the corresponding logging library package must be installed beforehand.

it needs to be reworded, right?

File: test/loginjection.test.ts (hunk 4 above)
Comment (Rauno56) [reply test/loginjection.test.ts:64]: This PR removes the option to toggle it. There's already a test for testing overwriting the default.

File: test/loginjection.test.ts (hunk 2 above)
Comment (Rauno56) [reply test/loginjection.test.ts]: Moved the default to signature.

**Comment by Rauno56** [comment]
> I unchecked the "documentation is done", because there's this sentence in the README:
> 
> > If log injection is enabled, the corresponding logging library package must be installed beforehand.
> 
> it needs to be reworded, right?

Yes, reworded it. Perhaps we want to remove it from the main readme altogether, @theletterf? There's more info in `docs/correlate-logs-traces.md`.

**Comment by theletterf** [comment]
@Rauno56 Fine by me!

File: examples/log-injection/index.js (hunk 1 above)
Comment (Rauno56) [reply examples/log-injection/index.js:28]: Our SDK currently doesn't provide access to the NodeTracerProvider we initialize.
#331

Comment (jtmal-signalfx) [reply examples/log-injection/index.js:28]: What about using API to get the global one?

**Review by jtmal-signalfx — APPROVED** [review]

File: examples/log-injection/index.js (hunk 1 above)
Comment (Rauno56) [reply examples/log-injection/index.js:28]: That currently gives us ProxyTracerProvider, which doesn't have forceFlush either and has NTP under private property. I don't have a full understanding of the correct way to solve this situation and that's why I'd like to leave it for now and figure it out after the next release.

**Pull request merged**